utilities/src/main/java/io/syndesis/qe/bdd/validation/MetricsValidationSteps.java

```diff
@@ -0,0 +1,60 @@
+package io.syndesis.qe.bdd.validation;
+
+import org.assertj.core.api.Assertions;
+
+import java.security.GeneralSecurityException;
+import java.util.Date;
+import java.util.List;
+
+import cucumber.api.java.en.Then;
+import io.syndesis.model.integration.Integration;
+import io.syndesis.model.metrics.IntegrationMetricsSummary;
+import io.syndesis.qe.endpoints.IntegrationsEndpoint;
+import io.syndesis.qe.endpoints.MetricsEndpoint;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class MetricsValidationSteps {
+
+	private IntegrationsEndpoint integrationsEndpoint;
+	private MetricsEndpoint metricsEndpoint;
+
+	public MetricsValidationSteps () {
+		try {
+			integrationsEndpoint = new IntegrationsEndpoint();
+			metricsEndpoint = new MetricsEndpoint();
+		} catch (GeneralSecurityException e) {
+			Assertions.fail("Error: " + e);
+		}
+
+
+	}
+	@Then("^validate that number of all messages through integration \"([^\"]*)\" is greater than \"([^\"]*)\", period in ms: \"([^\"]*)\"$")
+	public void validateThatNumberOfAllMessagesOfIntegrationIsGreaterThanPeriodInMs(String integrationName, Integer nr, Integer ms) throws InterruptedException {
+		Thread.sleep(ms + 1000);
+//		0.  get integration id.
+		String id = this.getIdByIntegrationName(integrationName);
+		Assertions.assertThat(id).isNotNull();
+
+//		1. get metrics:
+		IntegrationMetricsSummary summary = metricsEndpoint.get(id);
+//		2. get metrics info:
+		Long nrOfMessages = summary.getMessages().get();
+		log.info("MESSAGES SUMMARY: *{}*", nrOfMessages);
+		Assertions.assertThat(nrOfMessages).isEqualTo(3);
+
+		Date lastProceded = summary.getLastProcessed().get();
+		log.info("LAST MESSAGE WAS PROCEEDED ON: *{}*", lastProceded);
+	}
+
+
+//	AUXILIARIES
+
+	private String getIdByIntegrationName(String integrationName) {
+		List<Integration> integrations = integrationsEndpoint.list();
+		Integration integr = integrations.stream().filter(integration -> integrationName.equals(integration.getName())).findAny().orElse(null);
```

Comment:
Imho you could just put `return` here and it's fine to check `notNull` on line 37 as keep this as value-retrieving method only.

Reply:
done